s3/libirods_s3.cpp

```diff
@@ -589,15 +592,14 @@ irods::error s3InitPerOperation (
         const char* host_name = hostname.c_str(); // Iterate through on each try
         status = S3_initialize( "s3", flags, host_name );
 
-        std::stringstream msg;
-        if( status >= 0 ) {
-            msg << " - \"";
-            msg << S3_get_status_name((S3Status)status);
-            msg << "\"";
+        auto msg = fmt::format("[resource_name={}]  - Error initializing the S3 library. Status = {}.",
+                resource_name, status);
+
+        if(status >= 0) {
+            msg += fmt::format(" - \"{}\"", S3_get_status_name((S3Status)status));
         }
 
-        result = ASSERT_ERROR(status == S3StatusOK, status, "[resource_name=%s] Error initializing the S3 library. Status = %d.", resource_name.c_str(),
-                              status, msg.str().c_str());
+        result = ASSERT_ERROR(status == S3StatusOK, status, msg);
```

Comment:
We should avoid using these macros in new code / code changes.

Reply:
Understood but this will go in another pull request I believe.